GVFS/GVFS/CommandLine/GVFSVerb.cs

```diff
@@ -124,39 +124,112 @@ public static bool TrySetRequiredGitConfigSettings(Enlistment enlistment)
+
+                // Use 'multi-pack-index' builtin instead of 'midx' to match upstream implementation
                 { "core.multiPackIndex", "true" },
+
+                // Perf - Enable parallel index preload for operations like git diff
                 { "core.preloadIndex", "true" },
                 { "core.safecrlf", "false" },
+
+                // Possibly cause hydration while creating untrackedCache.
                 { "core.untrackedCache", "false" },
                 { "core.repositoryformatversion", "0" },
+
+                // Support file modes onr Mac & Linux.
                 { "core.filemode", GVFSPlatform.Instance.FileSystem.SupportsFileMode ? "true" : "false" },
+
+                // For consistency with git init. https://dev.azure.com/mseng/AzureDevOps/_git/GVFS/pullrequest/235679
                 { "core.bare", "false" },
+
+                // For consistency with git init. https://dev.azure.com/mseng/AzureDevOps/_git/GVFS/pullrequest/235679
                 { "core.logallrefupdates", "true" },
+
+                // Git to download objects on demand.
                 { GitConfigSetting.CoreVirtualizeObjectsName, "true" },
+
+                // Configure hook that git calls to ask GVFS abvout what fiels it cares about
                 { GitConfigSetting.CoreVirtualFileSystemName, Paths.ConvertPathToGitFormat(GVFSConstants.DotGit.Hooks.VirtualFileSystemPath) },
+
+                // Ensure hooks path is configured correctly.
                 { "core.hookspath", expectedHooksPath },
+
+                // Hostname is no longer sufficent for VSTS authentication. VSTS now requires dev.azure.com/account to determine the tenant.
+                // By setting useHttpPath, credential managers will get the path which contains the account as the first parameter. They can then use this information for auth appropriately.
                 { GitConfigSetting.CredentialUseHttpPath, "true" },
                 { "credential.validate", "false" },
+
+                // This setting is not needed anymore, because current version of gvfs does not use index.lock.
+                // (This change was introduced initially to prevent `git diff` from acquiring index.lock file.)
                 { "diff.autoRefreshIndex", "true" },
+
+                // Turn off of git garbage collection. Git garbage collection does not work with virtualized object. Check with Stolee - probably not needed.
                 { "gc.auto", "0" },
+
+                // Prevent git GUI from displaying a GC warning. https://mseng.visualstudio.com/AzureDevOps/_git/GVFS/pullrequest/184856?_a=overview
                 { "gui.gcwarning", "false" },
+
+                // Update git settings to enable optimizations in git 2.20
+                // Set 'index.threads=true' to enable multi-threaded index reads
                 { "index.threads", "true" },
+
+                // gvfs index parser would work only with index versions 4 and above.

```

Comment:
Suggested wording:
`index parsing code in VFSForGit currently only supports version 4`.